src/Rings/mpoly-graded.jl

```diff
@@ -1496,6 +1501,157 @@ function homogenization(f::MPolyElem, S::MPolyRing_dec, pos::Int = 1)
   return finish(B)
 end
 
+function _homogenization(f::MPolyElem, S::MPolyRing_dec, start_pos::Int = 1)
+   len_gg  = ngens(S.D)
+      #= get exponent vectors and enlarge them for S =#
+   exps = Vector{Int}[]
+   for e in exponent_vectors(f)
+       for j in 1:len_gg
+           insert!(e, start_pos+j-1, 0)
+       end
+       push!(exps, e)
+   end
+
+   #= compute degrees in S =#
+   if is_z_graded(S) == true
+       degs    = Int[]
+       gens_gg = [Int(S.d[i][1])  for i in 1:ngens(S)]
+   else
+       degs    = Vector{Int}[]
+       gens_gg = [[Int(S.d[i][j]) for j in 1:ngens(S.D)] for i in 1:ngens(S)]
+   end
```

Comment:
This code sets the variables `degs` and `gens_gg` to different types, depending on which code path is taken. This disables some type based optimizations. That may be completely fine in this function, just wanted to make you aware of it.